test/export/test_serialize.py

```diff
@@ -234,14 +234,14 @@ def check_graph(self, fn, inputs, constraints=None) -> None:
                     node2.meta.get("stack_trace", None),
                 )
 
-            # Check "nn_module_stack" metadata
-            self.assertEqual(
-                node1.meta.get("nn_module_stack", None),
-                node2.meta.get("nn_module_stack", None),
-            )
-
-            # Check "source_fn" metadata
-            if node1.op != "get_attr":
+            if node1.op != "get_attr" and node1.op != "placeholder":
```

Comment:
Should the deserializer then repopulate this information based on the param names? I'm a little hesitant for there to be different information before/after serialization

Reply:
Hmm since it is not in our contract to save nn_module_stack and source_fn in the graph node, we don't need to do it right now? The correct way would be downstream users use the param and buffer name to get the source_fn and nn_module_stack.

Reply:
Then why do we add it to the node here?

Reply:
This is simply to preserve the old flow where they still rely on graph node's metadata to do partition. I think once we actually make them use the param/buffer name to find the source_fn and nn_module_stack, we don't even need to do extra step of repopulating the graph node after deserializer.